contracts/modules/royalty/policies/LAP/RoyaltyPolicyLAP.sol

```diff
@@ -0,0 +1,321 @@
+        if (disputeModule == address(0)) revert Errors.RoyaltyPolicyLAP__ZeroDisputeModule();
+
+        ROYALTY_MODULE = IRoyaltyModule(royaltyModule);
+        DISPUTE_MODULE = IDisputeModule(disputeModule);
+        _disableInitializers();
+    }
+
+    /// @notice Initializer for this implementation contract
+    /// @param accessManager The address of the protocol admin roles contract
+    function initialize(address accessManager) external initializer {
+        if (accessManager == address(0)) revert Errors.RoyaltyPolicyLAP__ZeroAccessManager();
+        __ProtocolPausable_init(accessManager);
+        __ReentrancyGuard_init();
+        __UUPSUpgradeable_init();
+    }
+
+    /// @notice Executes royalty related logic on minting a license
+    /// @dev Enforced to be only callable by RoyaltyModule
+    /// @param ipId The ipId whose license is being minted (licensor)
+    /// @param licensePercent The license percentage of the license being minted
+    function onLicenseMinting(
+        address ipId,
+        uint32 licensePercent,
+        bytes calldata
+    ) external onlyRoyaltyModule nonReentrant {
+        // check if the new license royalty is within the royalty stack limit
+        if (_getRoyaltyStack(ipId) + licensePercent > ROYALTY_MODULE.TOTAL_RT_SUPPLY())
+            revert Errors.RoyaltyPolicyLAP__AboveRoyaltyStackLimit();
+    }
+
+    /// @notice Executes royalty related logic on linking to parents
+    /// @dev Enforced to be only callable by RoyaltyModule
+    /// @param ipId The children ipId that is being linked to parents
+    /// @param parentIpIds The parent ipIds that the children ipId is being linked to
+    /// @param licensesPercent The license percentage of the licenses being minted
+    function onLinkToParents(
+        address ipId,
+        address[] calldata parentIpIds,
+        address[] memory licenseRoyaltyPolicies,
+        uint32[] calldata licensesPercent,
+        bytes calldata
+    ) external onlyRoyaltyModule nonReentrant {
+        RoyaltyPolicyLAPStorage storage $ = _getRoyaltyPolicyLAPStorage();
+
+        uint32[] memory royaltiesGroupedByParent = new uint32[](parentIpIds.length);
+        address[] memory uniqueParents = new address[](parentIpIds.length);
+        uint256 uniqueParentCount;
+        for (uint256 i = 0; i < parentIpIds.length; i++) {
+            if (licenseRoyaltyPolicies[i] != address(this)) {
+                _setRoyaltyLAP(ipId, parentIpIds[i], 0);
+            } else {
+                (uint256 index, bool exists) = ArrayUtils.indexOf(uniqueParents, parentIpIds[i]);
+                if (!exists) {
+                    index = uniqueParentCount;
+                    uniqueParentCount++;
+                }
+                royaltiesGroupedByParent[index] += licensesPercent[i];
+                uniqueParents[index] = parentIpIds[i];
+                _setRoyaltyLAP(ipId, parentIpIds[i], royaltiesGroupedByParent[index]);
+            }
+        }
```

Comment:
Would be better adding more comments to explain the purpose of the code here.

Reply:
addressed here: https://github.com/storyprotocol/protocol-core-v1/pull/203/commits/c946aed9f5682dbc458e93b4f61c1cee07fee130